### PR TITLE
Update ProxyCheck

### PR DESCRIPTION
**Pull request by drknzz**
### **Please check if the PR fulfills these requirements**
- [x] The formatter, linter, and tests all run without an error
- [x] Tests for the changes have been added (for bug fixes/features)
- [ ] Docs have been added / updated (for bug fixes / features)

## **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Updates ProxyCheck subclasses.

Tests for OpenZeppelinProxyCheck and ArgentProxyCheck based on [OZ cairo contract](https://github.com/OpenZeppelin/cairo-contracts/blob/243df2b0935b09366ee08785c7cb3634c22c3642/src/openzeppelin/upgrades/presets/Proxy.cairo) and [Argent cairo contract](https://github.com/argentlabs/argent-contracts-starknet/blob/21466061765626824831d6304b9255b0d38338e7/contracts/Proxy.cairo)

## **What is the current behavior?** (You can also link to an open issue here)
Out of date ProxyCheck subclasses.

## **What is the new behavior (if this is a feature change)?**
OpenZeppelinProxyCheck subclasses are updated and pass the tests.

### CODE REVIEW

**Review by cptartur — COMMENTED** [review]
Would it be possible to use `pytest.mark.parametrize` instead of for loops in tests? Having for loops makes test outputs less readable since it's not immediately know which iteration failed.

**Review by war-in — APPROVED** [review]
Perfect! Just one question

File: starknet_py/tests/e2e/proxy_check/proxy_test.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+        client=gateway_account_client,
+    )
+    proxy2_deployment = await Contract.deploy(
+        compiled_contract=compiled_proxy,
+        constructor_args=[0x123],
+        client=gateway_account_client,
+    )
+    await proxy1_deployment.wait_for_acceptance()
+    await proxy2_deployment.wait_for_acceptance()
+
```
Comment (cptartur) [inline starknet_py/tests/e2e/proxy_check/proxy_test.py:53]: Here we wait for acceptance but in the above and below test we don't. We should wait in other cases as well.

File: starknet_py/tests/e2e/proxy_check/proxy_test.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        await proxy2.functions["_set_implementation"].invoke(
+            implementation=proxy1.address, max_fee=MAX_FEE
+        )
+    else:
+        await proxy1.functions["_set_implementation_hash"].invoke(
+            new_implementation=proxy2.address, max_fee=MAX_FEE
+        )
+        await proxy2.functions["_set_implementation_hash"].invoke(
+            new_implementation=proxy1.address, max_fee=MAX_FEE
+        )
```
Comment (cptartur) [inline starknet_py/tests/e2e/proxy_check/proxy_test.py:72]: Can this at least be abstracted into separate methods? It makes this test unnecessarily complicated.

**Review by cptartur — CHANGES_REQUESTED** [review]

File: starknet_py/tests/e2e/proxy_check/proxy_test.py [hunk 3]
```diff
@@ -33,6 +36,26 @@ async def test_argent_contract_from_address_throws_on_too_many_steps(
...
+        "proxy1": {implementation: proxy2.address},
+        "proxy2": {implementation: proxy1.address},
+    }
+
+    await proxy1.functions[set_implementation_name].invoke(
+        **params["proxy2"], max_fee=MAX_FEE
+    )
+    await proxy2.functions[set_implementation_name].invoke(
+        **params["proxy1"], max_fee=MAX_FEE
+    )
```
Comment (cptartur) [inline starknet_py/tests/e2e/proxy_check/proxy_test.py:56]: Check if these should be awaited

**Review by cptartur — APPROVED** [review]
Just a sigle comment, otherwise approved.

**Pull request merged**